### ISSUE TITLE
NPE in StreetLayer.getVertexIndexForOsmNode

### ISSUE THREAD

**Issue by demory** [conveyal/r5#240]
Running into the following bug trying to build the latest OSM for DC. The OSM we're using is on S3 in the following bundle: https://s3.amazonaws.com/datatools-prod/bundles/b6f4edc5-87ce-42bf-9acd-635dd66f7f71/02-20-17+Bundle.zip

```[main] WARN com.conveyal.r5.labeling.LevelOfTrafficStressLabeler - Unable to parse maxspeed tag unposted
Exception in thread "main" java.lang.NullPointerException
        at com.conveyal.r5.streets.StreetLayer.getVertexIndexForOsmNode(StreetLayer.java:883)
        at com.conveyal.r5.streets.StreetLayer.makeEdge(StreetLayer.java:927)
        at com.conveyal.r5.streets.StreetLayer.loadFromOsm(StreetLayer.java:264)
        at com.conveyal.r5.streets.StreetLayer.loadFromOsm(StreetLayer.java:170)
        at com.conveyal.r5.transit.TransportNetwork.fromFiles(TransportNetwork.java:162)
        at com.conveyal.r5.transit.TransportNetwork.fromFiles(TransportNetwork.java:218)
        at com.conveyal.r5.transit.TransportNetwork.fromDirectory(TransportNetwork.java:252)
        at com.conveyal.r5.point_to_point.PointToPointRouterServer.main(PointToPointRouterServer.java:77)
        at com.conveyal.r5.R5Main.main(R5Main.java:33)```

**Comment by buma** [comment]
I have tried to reproduce it, but I get error 403 forbidden when trying to download data from S3. The only way that this could happen is that there is a way in OSM with nodes that aren't in same OSM data.

**Comment by buma** [comment]
The problem is that OSM is missing this [node](https://www.openstreetmap.org/node/4695821252) which is first node in way [this 475902160](https://www.openstreetmap.org/way/475902160). This shouldn't happen and it seems that there was some problem generating this OSM.

**Comment by abyrd** [comment]
Is this data pulled from OSM-lib (VEX)?

**Comment by abyrd** [comment]
It sounds like this was caused by OSM data containing ways referencing nodes that are not included in the OSM data. The simplest solution is to provide input data with referential integrity. When working with data that have been cropped with Osmosis, providing the completeWays=yes parameter to the Osmosis bounding-box operation will ensure that all referenced nodes are included in the clipped OSM file.

This issue should be resolved by commit a55ffc3, see (duplicate) issue #276.
